dts/arm/nxp/nxp_mcxn94x_common.dtsi

```diff
@@ -618,6 +618,22 @@
 		clk-divider = <1>;
 	};
 
+	flexio0: flexio@105000 {
+		compatible = "nxp,flexio";
+		reg = <0x105000 0x1000>;
+		status = "disabled";
+		interrupts = <105 0>;
+		clocks = <&syscon MCUX_FLEXIO0_CLK>;
+		flexio0_lcd: flexio0-lcd {
+			compatible = "nxp,mipi-dbi-flexio-lcdif";
+			status = "disabled";
+		};
+		flexio0_spi0: flexio0_spi0 {
```

Comment:
Why add the SPI node? Is this needed for this PR? I mainly ask since we don't seem to enable this anywhere, so this isn't really being tested

Reply:
I chose to add the flexio-spi compatible so that we are aware this support exists and it can be enabled in future.

Reply:
I'd prefer we add this in a separate PR- generally if we define a node with a given compatible in an SOC devicetree, I think the best practice is to have at least one test in tree that is set up to verify that the device defined by the node actually functions. So if we define the flexio_spi node, we should also add an overlay to the spi loopback test to enable testing this functionality